engine/common/rpc/convert/execution_results.go

```diff
@@ -181,20 +181,35 @@ func MessageToChunk(m *entities.Chunk) (*flow.Chunk, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to parse Message end state to Chunk: %w", err)
 	}
-	chunkBody := flow.ChunkBody{
-		CollectionIndex:      uint(m.CollectionIndex),
-		StartState:           startState,
-		EventCollection:      MessageToIdentifier(m.EventCollection),
-		BlockID:              MessageToIdentifier(m.BlockId),
-		TotalComputationUsed: m.TotalComputationUsed,
-		NumberOfTransactions: uint64(m.NumberOfTransactions),
-		ServiceEventCount:    MessageToServiceEventCountField(m.ServiceEventCount),
+
+	serviceEventCountPtr := MessageToServiceEventCountField(m.ServiceEventCount)
+
+	// Branch on nil-vs-non-nil to preserve backward compatibility
+	if serviceEventCountPtr == nil {
+		// Protocol v1: omit ServiceEventCount
+		return flow.NewChunk_ProtocolVersion1(
```

Comment:
I thought that we have agreed to use Untrusted**Type** for passing to constructor to avoid accidental reordering of arguments, haven't we?